### PR TITLE
Initial viewer implementation

### PR DESCRIPTION
**Pull request by caffodian**
Given a page with a bunch of headings w/ ids, create the convenient clipboard links beside them.  

todo in a followup:
- handle failures with the popover
- pad the share thingy better & other styling tweaks 
- have a better way of running the test runner (right now you just open it in a browser, npm test doesn't do anything) 
- grab testrunner dependencies from bower/npm instead of cdn
- figure out how to run these in travis (separate from the bender stuff)

### CODE REVIEW

**Comment by caffodian** [comment]
screen shot 2016-11-15 at 10 57 47 am

the test runner page

File: viewer/test/test_add_heading_anchors.js [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+                    clipboardDataAttribute;
+
+                // is element
+                assert.equal(anchor.nodeType, 1);
+
+                // is an anchor
+                assert.equal(anchor.nodeName, 'A');
+
+                // href is to id
+
```
Comment (winhamwr) [inline viewer/test/test_add_heading_anchors.js]: Why the blank line, here? Ditto a few lines down.

File: viewer/testrunner.html [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        <h5>No-id Heading 5</h5>
+        <h6>No-id Heading 6</h6>
+    </div>
+    <div id="mocha"></div>
+    <script src="node_modules/mocha/mocha.js"></script>
+    <script src="node_modules/chai/chai.js"></script>
+    <script src="node_modules/chai-string/chai-string.js"></script>
+    <script>mocha.setup('bdd')</script>
+
+    <!-- load code you want to test here -->
```
Comment (winhamwr) [inline viewer/testrunner.html]: Remove template comment

File: viewer/testrunner.html [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    <script src="https://code.jquery.com/jquery-1.12.4.js" integrity="sha256-Qw82+bXyGq6MydymqBxNPYTaUXXq7c8v3CwiYwLLNXU=" crossorigin="anonymous"></script>
+    <script src="https://maxcdn.bootstrapcdn.com/twitter-bootstrap/2.3.2/js/bootstrap.min.js" integrity="sha384-vOWIrgFbxIPzY09VArRHMsxned7WiY6hzIPtAIIeTFuii9y3Cr6HE6fcHXy5CFhc" crossorigin="anonymous"></script>
+    <script src="node_modules/clipboard/dist/clipboard.min.js"></script>
+
+    <script src='add_heading_anchors.js'></script>
+
+    <!-- load your test files here -->
```
Comment (winhamwr) [inline viewer/testrunner.html]: Remove template comment

**Review by winhamwr — APPROVED** [review]
Looks like good progress. All of my thoughts are covered by the follow-on spec in the PR description. 

Could you run a linter on this and give it a walk through for code style? There are a lot of things separated by blank lines that don't seem to be semantic.

**Comment by caffodian** [comment]
I tried out VS Code for this one, and I don't have a good understanding of what the autoformat does.  Playing with eslint fixing, which will blow up the diff.  

I will comment that the blank lines around comment is an (over-configurable) thing:  http://eslint.org/docs/rules/lines-around-comment :)

**Comment by caffodian** [comment]
After eslint stuff (pro tip:  update eslint extension in VS Code if you run into bizarre behaviour) , tests still run.  Yay

**Pull request merged**